packages/itwin/grouping-mapping-widget/src/components/Mappings/hooks/useFetchMappingExtractionStatus.ts

```diff
@@ -0,0 +1,102 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import type { QueryClient } from "@tanstack/react-query";
+import { useQuery } from "@tanstack/react-query";
+import type { ExtractionStatus, IExtractionClient } from "@itwin/insights-client";
+import { ExtractorState } from "@itwin/insights-client";
+import type { Mapping } from "@itwin/insights-client";
+import { STATUS_CHECK_INTERVAL } from "../../Constants";
+import { ExtractionStates } from "../Extraction/ExtractionStatus";
+import type { GetAccessTokenFn, GroupingMappingApiConfig } from "../../context/GroupingApiConfigContext";
+import { useExtractionClient } from "../../context/ExtractionClientContext";
+import { useMemo } from "react";
+import { useExtractionStateJobContext } from "../../context/ExtractionStateJobContext";
+
+export interface MappingExtractionStatusProps extends GroupingMappingApiConfig {
+  isMounted: boolean;
+  mapping: Mapping;
+  enabled: boolean;
+}
+
+export interface MappingQueryResults {
```

Comment:
I think it would look neater if you defined it as `MappingQueryResult` and then used `MappingQueryResult[]` where needed.

Reply:
I have changed it to `MappingQueryResult`.